auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue

```diff
@@ -130,134 +150,163 @@
+        this.$router.push(Pages.STAFF_DASHBOARD)
+      }
     }
-  }
 
-  @Watch('reviewResult')
-  @Watch('emailBodyText')
-  onDataChanged (): void {
-    if (this.reviewResult || this.emailBodyText) {
-      this.haveUnsavedChanges = true
+    /** Called when user clicks Cancel button. */
+    function onClickCancel (): void {
+      state.haveUnsavedChanges = false
+      // route back to staff dashboard
+      this.$router.push(Pages.STAFF_DASHBOARD)
     }
-  }
 
-  onClickCancel (): void {
-    this.haveUnsavedChanges = false
-    this.$router.push(Pages.STAFF_DASHBOARD)
-  }
-
-  async onClickSubmit (): Promise<void> {
-    if (!this.$refs.reviewResult.validate()) return
-
-    try {
-      this.isSubmitting = true
-      await BusinessService.saveContinuationReviewResult(this.reviewResult, this.emailBodyText)
-      this.haveUnsavedChanges = false
-
-      // indicate success via toast (snackbar) on main page
-      EventBus.$emit('show-toast', {
-        message: 'Review submitted successfully',
-        type: 'primary',
-        timeout: 3000
-      })
-    } catch (error) {
-      // eslint-disable-line no-console
-      console.log(`Error submitting review = ${error}`)
-
-      // show error dialog
-      this.dialogTitle = 'Unable to submit review'
-      this.dialogText = 'An error occurred while submitting the review. Please try again.'
-      this.$refs.errorDialog.open()
-    } finally {
-      this.isSubmitting = false
+    /** Called when user clicks Submit button. */
+    async function onClickSubmit (): Promise<void> {
+      // check component validity
+      if (!(reviewResultComponent as any).value.validate()) return
+
+      try {
+        state.isSubmitting = true
+        await BusinessService.submitContinuationReviewResult(state.review.id, state.reviewResult, state.emailBodyText)
+        state.haveUnsavedChanges = false
+
+        // route back to staff dashboard
+        this.$router.push(Pages.STAFF_DASHBOARD)
+
+        // indicate success via toast (snackbar) on main page
+        EventBus.$emit('show-toast', {
+          message: 'Review submitted successfully',
+          type: 'primary',
+          timeout: 3000
+        })
+      } catch (error) {
+        console.log(`Error submitting review = ${error}`) // eslint-disable-line no-console
+
+        // show error dialog
+        state.dialogTitle = 'Unable to submit review'
+        state.dialogText = 'An error occurred while submitting the review. Please try again.'
+        const v = errorDialogComponent.value as any; v.open()
+      } finally {
+        state.isSubmitting = false
+      }
     }
 
-    // if submit succeeded, route back to staff dashboard
-    if (!this.haveUnsavedChanges) this.$router.push(Pages.STAFF_DASHBOARD)
-  }
-
-  /** Intercepts route change (but not page unloads). */
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  async beforeRouteLeave (to, from, next): Promise<void> {
```

Comment:
Travis, I wasn't able to replicate this in Composition API. I googled for example and didn't find any, and '@vue/composition-api' didn't provide a similar lifecycle hook.

How would you deal with this?

Reply:
![image](https://github.com/user-attachments/assets/180470de-24ed-4f3e-ae55-504715a7aca7)

Reply:
https://router.vuejs.org/guide/advanced/composition-api.html#navigation-guards

Reply:
Thanks. Looking...

Reply:
The example above didn't work in my component, but I came up with something that works as desired. Please have a look. Resolving.